torch/nested/_internal/nested_tensor.py

```diff
@@ -161,6 +160,27 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
 
         raise NotImplementedError(func)
 
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+        # Handle SDPA specially since it's CompositeImplicit. We don't want
+        # the nestedness of the inputs to affect the kernel choice, so unwrap
+        # the NTs here before passing to SDPA -> rewrap the output as NT.
+        if func is torch._C._nn.scaled_dot_product_attention:
+            t_args = [t._values if isinstance(t, NestedTensor) else t for t in args]
+            t_kwargs = {
+                k: v._values if isinstance(v, NestedTensor) else v
+                for k, v in kwargs.items()
+            }
+
+            from torch.nested._internal.ops import extract_kwargs
+
+            output = func(*t_args, **t_kwargs)
+            return NestedTensor(output, **extract_kwargs(args[0]))
+        with torch._C.DisableTorchFunctionSubclass():
+            return func(*args, **kwargs)
```

Comment:
What's the plan here with torch compile?

Reply:
(I suspect tests would fail)

Reply:
Yeah good question, I need to look into this. For the purposes of getting SAM running without torch.compile, this is fine, but not great for the immediate next step